packages/SwingSet/test/virtualObjects/vat-weakcollections-alice.js

```diff
@@ -0,0 +1,69 @@
+/* global makeKind */
+import { Far } from '@agoric/marshal';
+
+function makeHolderInstance(state) {
+  function init(value) {
+    state.value = value;
+  }
+  const self = Far('holder-vo', {
+    getValue() {
+      return state.value;
+    },
+    setValue(newValue) {
+      state.value = newValue;
+    },
+  });
+  return { init, self };
+}
+
+const holderMaker = makeKind(makeHolderInstance);
+
+export function buildRootObject() {
+  const testWeakMap = new WeakMap();
+  let memorableExportRemotable;
+  let memorableExportVirtualObject;
+  let memorableExportPromise;
+  let forgetableExportRemotable;
```

Comment:
To be even more explicit, you could put the `forgetable` ones inside the scope of `prepareWeakMap`, so they wouldn't even be in scope during `probeWeakMap`. Nulling them out is fine belt-and-suspenders. (I can imagine a JS engine following a different code path for "value fell out of scope" vs "value was explicitly overwritten", with conceivably different GC timing).